tests/conftest.py

```diff
@@ -608,3 +611,33 @@ async def setup_sim():
     sim_client = SimClient(sim)
     await sim.farm_block()
     return sim, sim_client
+
+
+@pytest.fixture(scope="function")
+def tmp_chia_root(tmp_path):
+    """
+    Create a temp directory and populate it with an empty chia_root directory.
+    """
+    path: Path = tmp_path / "chia_root"
+    path.mkdir(parents=True, exist_ok=True)
+    return path
+
+
+@pytest.fixture(scope="function")
+def root_path_populated_with_config(tmp_chia_root) -> Path:
+    """
+    Create a temp chia_root directory and populate it with a default config.yaml.
+    Returns the chia_root path.
+    """
+    root_path: Path = tmp_chia_root
+    create_default_chia_config(root_path)
+    return root_path
+
+
+@pytest.fixture(scope="function")
+def config_with_address_prefix(root_path_populated_with_config: Path, prefix: str) -> Iterator[Dict[str, Any]]:
+    updated_config: Dict[str, Any] = {}
+    with lock_and_load_config(root_path_populated_with_config, "config.yaml") as config:
+        if prefix is not None:
+            config["network_overrides"]["config"][config["selected_network"]]["address_prefix"] = prefix
+        yield config
```

Comment:
Now I have to apologize.  I looked back at my comment and I did use the word yield though I meant it abstractly, not as a Python `yield`...  that was a terrible choice of word.  The issue here is that the config file will be 'locked' in terms of the load functions for the entire test which will probably confuse someone at some point and be a problem.

```suggestion
@pytest.fixture(scope="function")
def config_with_address_prefix(root_path_populated_with_config: Path, prefix: str) -> Dict[str, Any]:
    updated_config: Dict[str, Any] = {}
    with lock_and_load_config(root_path_populated_with_config, "config.yaml") as config:
        if prefix is not None:
            config["network_overrides"]["config"][config["selected_network"]]["address_prefix"] = prefix
    return config
```